public/js/notifications.js

```diff
@@ -0,0 +1,433 @@
+                            if (this.eventSource !== null && this.eventSource.readyState !== EventSource.CLOSED) {
+                                this.eventSource.close();
+                            }
+
+                            break;
+                    }
+                }
+            });
+
+            // register service worker if it is not already
+            this.getServiceWorker()
+                .then((serviceWorker) => {
+                    if (! serviceWorker) {
+                        // no service worker registered yet, registering it
+                        self.navigator.serviceWorker.register(icinga.config.baseUrl + '/notifications-worker.js', {
+                            scope: icinga.config.baseUrl + '/',
+                            type: 'classic'
+                        }).then((registration) => {
+                            let callback = (event) => {
+                                console.log("YEP");
+                                if (event.target.state === 'activated') {
+                                    registration.removeEventListener('statechange', callback);
+
+                                    registration.active.postMessage(
+                                        JSON.stringify({
+                                            command: 'handshake',
+                                            version: VERSION
+                                        })
+                                    );
+                                }
+                            };
+                            registration.addEventListener('statechange', callback);
+                        });
+                    } else {
+                        // service worker is already running, announcing ourselves
+                        serviceWorker.postMessage(
+                            JSON.stringify({
+                                command: 'handshake',
+                                version: VERSION
+                            })
+                        )
+                    }
+                })
+                .finally(() => {
+                    this.logger.debug(this.prefix + "loaded");
+                })
+        }
+
+        unload () {
+            this.logger.debug(this.prefix + "unloading");
+
+            // disconnect EventSource if there's an active connection
+            if (this.eventSource && this.eventSource instanceof EventSource) {
+                if (this.eventSource.readyState !== EventSource.CLOSED) {
+                    this.eventSource.close();
+                }
+            }
+            this.eventSource = null;
+            this.initialized = false;
+
+            this.logger.debug(this.prefix + "unloaded");
+        }
+
+        reload () {
+            this.unload();
+            this.load();
+        }
+
+        openEventStream () {
+            if (this.hasNotificationPermission() && this.hasNotificationsEnabled()) {
```

Comment:
I would flip the if, else condition and return earlier.
```suggestion
            if (! this.hasNotificationPermission() || ! this.hasNotificationsEnabled()) {
             // the log from else condition
             
                return;
            }
            
            // if condition body
```